### PR TITLE
SUBMARINE-267. Initial job server which defines REST skeleton

### PR DESCRIPTION
**Pull request by tangzhankun**
### What is this PR for?
The initial job server which defines REST skeleton

### What type of PR is it?
Feature 

### Todos
* [1] - Fix the YAML unit test case issue
* [2] - Populate the REST API with integration to yarn-submitter or k8s-submitter
* [3] - Add Swagger to easy access the API spec

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-267

### How should this be tested?
CI unit test is enough for the REST API before we integrate with submitter

### Questions:
* Does the licenses files need update? No
* Are there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-server/server-core/src/test/java/JobApiTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        .request()
+        .post(Entity.entity(jobSpec, MediaType.APPLICATION_JSON));
+
+    assertEquals("Http Response should be 202 ",
+        Response.Status.ACCEPTED.getStatusCode(), response.getStatus());
+  }
+
+  // Test job created with incorrect JSON input
+  @Test
+  public void testCreateJobWhenJsonInputIsWrongThenResponseCodeAccepted() {
```
Comment (pingsutw) [inline submarine-server/server-core/src/test/java/JobApiTest.java]: would it be better to change to `testCreateJobWhenJsonInputIsWrongThenResponseCodeBadRequest` ?

**Comment by yuanzac** [comment]
@tangzhankun, Thanks for your contribution.
The test is failed. Can you take a look?

File: submarine-server/server-core/src/test/java/JobApiTest.java (hunk 1 above)
Comment (tangzhankun) [reply submarine-server/server-core/src/test/java/JobApiTest.java]: Good catch. Will change it.

File: submarine-server/server-core/src/main/java/org/apache/submarine/jobserver/JobServer.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    apiServlet.setInitParameter("jersey.config.server.provider.packages",
+        "org.apache.submarine.jobserver.rest");
+
+    try {
+      jobServer.start();
+      LOG.info("Submarine job server started");
+      jobServer.join();
+    } catch (Exception e) {
+      LOG.info("Submarine job server failed to start");
+      e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (tangzhankun) [reply submarine-server/server-core/src/main/java/org/apache/submarine/jobserver/JobServer.java]: Good catch! Will change it.

**Comment by tangzhankun** [comment]
@yuanzac @jiwq , my local build doesn't have this dependency convergence issue. Could you please help to clone the code and "mvn clean package -DskipTests"?

File: .travis.yml [hunk 3]
```diff
@@ -160,6 +160,13 @@ matrix:
         - npm run e2e -- --protractor-config=e2e/protractor-ci.conf.js
       env: NAME="Build workbench-web-ng"
 
+
+    # Test submarine-server
+    - language: java
+      jdk: "openjdk8"
+      dist: xenial
+      env: NAME="Test submarine-server" PROFILE="-Phadoop-2.9" BUILD_FLAG="clean package install -DskipTests" TEST_FLAG="test -DskipRat -am" MODULES="-pl ${EXCLUDE_WORKBENCH},${EXCLUDE_INTERPRETER}" TEST_MODULES="-pl submarine-server/server-core" TEST_PROJECTS=""
```
Comment (yuanzac) [inline .travis.yml]: There is another test named "Test submarine-server" in line 107 as well. 
It looks like their functionalities are the same. Can we combine them?

Comment (liuxunorg) [reply .travis.yml]: @yuanzac line 107 is "Test workbench-server". Not is "Test submarine-server".

**Pull request closed without merging**